src/main/java/application/SessionController.java

```diff
@@ -274,6 +288,12 @@ public String handleError(HttpServletRequest req, Exception exception) {
         return errorReturn.toString();
     }
 
+    public Lock getLockAndLock(SessionBean sessionBean){
```

Comment:
nice this makes sense to lock on session id. not sure if java has a decorator concept but this would be a perfect decorator. or maybe we make a response subclass that automatically does this